### PR TITLE
feat(date-picker): Add date-picker

### PR DESCRIPTION
**Pull request by ArthusLiang**
create new component.

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [x ] My code follows the code style of this project.
- [x ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

### CODE REVIEW

File: angular/src/docs/app/playground/playground.component.ts [hunk 1]
```diff
@@ -5,10 +5,7 @@ import { Component, OnInit } from '@angular/core';
   template: `
     <div className="row">
       <div className="example-spacing">
-          <example-alert-default></example-alert-default>
-          <example-alert-success></example-alert-success>
-          <example-alert-warning></example-alert-warning>
-          <example-alert-error></example-alert-error>
+          <example-date-picker-default></example-date-picker-default>
```
Comment (daanishnasir) [inline angular/src/docs/app/playground/playground.component.ts]: let's try not to push up the playground.ts; I often forgot to exclude it myself.

File: angular/src/lib/date-picker/date-picker-calendar.component.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  public allPrevDaysDisabled:boolean;
+  public allNextDaysDisabled:boolean;
+  public previousArialLabel:string;
+  public nextArialLabel:string;
+  public offsets = [0,1,2,3,4,5,6];
+  public today;
+
+  @HostBinding('class') classes = 'cui-datepicker__month-container';
+
+  //@HostListener('keyup', ['$event']) moveFocus($event) {
```
Comment (daanishnasir) [inline angular/src/lib/date-picker/date-picker-calendar.component.ts]: can remove the comment?

File: angular/src/lib/date-picker/date-picker-day.component.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    return 'cui-button'+
+    ' cui-button--circle cui-button--28' +
+    `${(this.isDisable && ` cui-button--disabled`) || ''}` +
+    ' cui-datepicker__day'+
+    `${(this.isSelected && ` cui-datepicker__day--selected`) || ''}` +
+    `${(this.isFocus && ` cui-datepicker__day--focus`) || ''}` +
+    `${(this.isToday && ` cui-datepicker__day--today`) || ''}` +
+    `${(this.isOutsideMonth && ` cui-datepicker__day--outside-month`) || ''}` +
+    ``;
+  };
```
Comment (daanishnasir) [inline angular/src/lib/date-picker/date-picker-day.component.ts]: So i believe we are going to use the updated class pattern here and apply the static css classes as well as boolean css to the host inside @Component({}). @pauljeter

File: angular/src/lib/date-picker/date-picker-month.component.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  styles: [
+    '.cui-datepicker__week{ display: block !important; }'
+  ]
+})
+export class DatePickerMonthComponent implements OnInit {
+
+  public weeks=[];
+  public month:any;
+
+  @HostBinding('class') classes = 'cui-datepicker__month';
```
Comment (daanishnasir) [inline angular/src/lib/date-picker/date-picker-month.component.ts]: Move to host:

File: angular/src/lib/date-picker/date-picker-week.component.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  /** @prop Required day that the DatePickerDay displays */
+  @Input() public day:any;
+  @Input() public month:any;
+
+  public days=[];
+
+  @HostBinding('class') classes:string = 'cui-datepicker__week';
+
+  constructor(
+    private datePickerService:DatePickerService
```
Comment (daanishnasir) [inline angular/src/lib/date-picker/date-picker-week.component.ts]: can all be in 1 line

File: angular/src/lib/date-picker/date-picker.component.ts [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+  @Input() public overlayY: VerticalConnectionPos = 'bottom';
+
+  /** @prop Handler invoked when user makes a change within the DatePicker | null */
+  @Output() public whenChange = new EventEmitter();
+  /** @prop Handler invoked when user makes a change to the selected month within DatePicker | null */
+  @Output() public whenMonthChange = new EventEmitter();
+  /** @prop Handler invoked when user selects a date within DatePicker | null */
+  @Output() public whenSelect = new EventEmitter();
+  
+  @HostBinding('class') get classes(): string {
```
Comment (daanishnasir) [inline angular/src/lib/date-picker/date-picker.component.ts]: to host:

File: angular/src/lib/date-picker/date-picker-calendar.component.ts (hunk 2 above)
Comment (ArthusLiang) [reply angular/src/lib/date-picker/date-picker-calendar.component.ts]: Sure, I will submit a change after the css file is updated.

**Pull request closed without merging**